src/msg/simple/Pipe.cc

```diff
@@ -2461,7 +2461,7 @@ int Pipe::tcp_read(char *buf, int len)
     buf += got;
     //lgeneric_dout(cct, DBL) << "tcp_read got " << got << ", " << len << " left" << dendl;
   }
-  return len;
+  return 0;
```

Comment:
Ah, thanks for this explanation. So it was a "bug" because it returned ```len``` before?
I often see such stuff in c code. Functions that return success or error but use ```int``` to represent this. Why the return type is not bool?